### PR TITLE
OH2-396 | Add a lock mechanism in all the remaining entities without it

### PR DESCRIPTION
**Pull request by SilverD3**
See OH2-396

I skipped `MedicalWard` as many tests are failing when I add the `lock` (`@Version`) property in that entity. I investigated and I realized that we need to refactor a lot of code `MedicalStokcIoOperations` to make `MedicalWard` entity support `lock` mechanism.

So if it's really needed, we'll open another issue for that.

### CODE REVIEW

File: sql/step_a112_add_missing_lock_columns.sql
Comment (mwithi) [inline sql/step_a112_add_missing_lock_columns.sql:1]: column names are UPPERCASE in this project

Comment (SilverD3) [reply sql/step_a112_add_missing_lock_columns.sql:1]: They're stored in lowercase in the DBMS, but okay. Let's use uppercase

Comment (mwithi) [reply sql/step_a112_add_missing_lock_columns.sql:1]: table names are lowercase (if `lower_case_table_names = 1`) but fields should be created as specified in the script

**Comment by mwithi** [comment]
> So if it's really needed, we'll open another issue for that.

Created OH2-409

File: sql/step_a112_add_missing_lock_columns.sql
Comment (mwithi) [inline sql/step_a112_add_missing_lock_columns.sql:1]: missing changes on step_04 and demo_data scripts

File: sql/step_a112_add_missing_lock_columns.sql [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ALTER TABLE `oh_pricesothers`
+    ADD COLUMN `OTH_LOCK` INT(11) NOT NULL DEFAULT 0;
+
+ALTER TABLE `oh_visits`
+    ADD COLUMN `VST_LOCK` INT(11) NOT NULL DEFAULT 0;
+
+
+
+
+
```
Comment (mwithi) [inline sql/step_a112_add_missing_lock_columns.sql]: Please remove trailing empty lines

File: sql/step_04_all_following_steps.sql [hunk 2]
```diff
@@ -104,3 +104,4 @@ source step_a107_add_permission_admin_access.sql;
 source step_a108_audit_dicom_type_and_dicom_data.sql;
 source step_a109_update_user_settings_table_constraints_and_add_usergroups_permissions.sql;
 source step_a110_update_operations_table_change_ope_for_to_enum.sql;
+source step_a112_add_missing_lock_columns.sql;
```
Comment (mwithi) [inline sql/step_04_all_following_steps.sql]: Please rename it as probably been merged before OP-1310
```suggestion
source step_a111_add_missing_lock_columns.sql;
```

File: sql/step_a112_add_missing_lock_columns.sql
Comment (mwithi) [inline sql/step_a112_add_missing_lock_columns.sql:1]: Rename to `step_a111_add_missing_lock_columns.sql`

**Review by mwithi — CHANGES_REQUESTED** [review]
Add some comments, and waiting for updated `load_data_demo.sql`

**Review by mwithi — APPROVED** [review]

**Pull request merged**